src/parser.ts

```diff
@@ -966,7 +968,7 @@ export class Parser extends DiagnosticEmitter {
         }
       }
     }
-    if (typeParameters.length === 0) {
+    if (!typeParameters || typeParameters.length === 0) {
```

Comment:
nit: `if (!(typeParameters && typeParameters.length))` maybe. Saves an i32.const and I find it easier to grasp.

Reply:
makes sense